### ISSUE TITLE
Any way to customize facet position?

### ISSUE THREAD

**Issue by thiagoveloso** [tidyverse/ggplot2#4179]
I am plotting a `ggplot2` graph that has an uneven number of facets. My plot currently looks something like this:

```
library(ggplot2)

df <- data.frame(group=c(1,1,2,2,3),
                 name=c('a','b','c','d','e'),
                 x=c(1,2,3,4,5),
                 y=c(2,3,4,5,6))
ggplot(df, aes(x,y)) + geom_point() + facet_wrap(~ name, ncol=3) 
```

[![enter image description here][1]][1]

Note that the default positioning for the facets is:

```
a b c
d e
```

However, I would like to have a finer control on the positioning of the facets. Specifically, I would like to "center" the bottom two facets in order to distribute the void space more evenly on the sides of the plot.

In other words, the ideal placement for me would look like:

```
a b c
 d e
```

Is this something that can be achieved with `ggplot2`? How? Many thanks in advance.

  [1]: https://i.stack.imgur.com/Qkq50.png

**Comment by karawoo** [comment]
There is not a way to do this purely in ggplot2, but you could use [patchwork](https://github.com/thomasp85/patchwork) or [cowplot](https://github.com/wilkelab/cowplot). For future reference, this type of question is a better fit for [RStudio Community](https://community.rstudio.com/).
